### PR TITLE
added support for new g1-rfc9380 scheme

### PR DESCRIPTION
**Pull request by CluEleSsUK**

### CODE REVIEW

File: lib/beacon-verification.ts [hunk 1]
```diff
@@ -48,13 +53,19 @@ function normP2(point: G2Hex): PointG2 {
...
+    return point instanceof PointG1 ? point : PointG1.hashToCurve(point, {DST: domainSeparationTag});
 }
 
-export async function verifySigOnG1(signature: G1Hex, message: G1Hex, publicKey: G2Hex): Promise<boolean> {
+export async function verifySigOnG1(
+    signature: G1Hex,
+    message: G1Hex,
+    publicKey: G2Hex,
+    // default DST is the invalid one used for 'bls-unchained-on-g1' for backwards compat
+    domainSeparationTag= 'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_'
```
Comment (AnomalRoil) [inline lib/beacon-verification.ts:65]: Gosh, I hate this. But I guess it's the only way not to break things for people T.T
Are we sure people are actually using this function directly?
Could we avoid doing this? In the code above we could be explicit for each scheme...

**Review by AnomalRoil — APPROVED** [review]
LGTM I guess, but would be nice to avoid carrying that tech debt of retro-compat of non-compliant verification forever.

File: lib/beacon-verification.ts (hunk 1 above)
Comment (CluEleSsUK) [reply lib/beacon-verification.ts:65]: we could make it mandatory, but we'd have to bump to v2 to make sure nobody gets pwned by it

Comment (CluEleSsUK) [reply lib/beacon-verification.ts:65]: Have raised an issue here: https://github.com/drand/drand-client/issues/70

**Pull request merged**